tests/ui/ui_tester.py

```diff
@@ -0,0 +1,173 @@
+            self._driver.window_handles[len(self._driver.window_handles) - 1]
+        )
+
+    def output_images(self, selector, num_of_screenshots=10):
+        output_dir = self._get_output_dir('out')
+        element = "document.getElementById('{}')".format(selector)
+        scroll_height = float(self._driver.execute_script('return {}.scrollHeight'.format(element)))
+        for i in range(num_of_screenshots):
+            y = (i / num_of_screenshots) * scroll_height
+            self._driver.execute_script('{}.scrollTo(0, {})'.format(element, y))
+            self.save_screenshot(output_dir, '{}{}.png'.format(self._test_name, i + 1))
+
+    def compare_to_baselines(self, num_of_screenshots=10):
+        out_dir = self._get_output_dir('out')
+        baselines_dir = self._get_output_dir('baselines')
+        diff_dir = self._get_output_dir('diff')
+        has_failed = False
+        for i in range(num_of_screenshots):
+            screenshot_filename = '{}{}.png'.format(self._test_name, i + 1)
+            out_img = os.path.join(out_dir, screenshot_filename)
+            baseline_img = os.path.join(baselines_dir, screenshot_filename)
+            diff_img = os.path.join(diff_dir, screenshot_filename)
+            diff_percent = (
+                diff(baseline_img, out_img, delete_diff_file=True, ignore_alpha=True) * 100
+            )
+            if diff_percent > UITester._SCREENSHOT_DIFF_THRESHOLD_PERCENT:
+                # If an image comparison has failed, generate diff and print an error message in red
+                has_failed = True
+                diff(
+                    out_img,
+                    baseline_img,
+                    delete_diff_file=False,
+                    diff_img_file=diff_img,
+                    ignore_alpha=True,
+                )
+                print(
+                    '\033[91mScreenshot comparison failed in {} for {} by {}%\033[0m'.format(
+                        self._get_browser(), screenshot_filename, diff_percent
+                    )
+                )
+
+        assert not has_failed
+
+    def get_url(self, path):
+        return '{}/{}'.format(self._base_url, path)
+
+    def _get_output_dir(self, folder_name):
+        def create_path(path):
+            if not os.path.isdir(path):
+                os.mkdir(path)
+
+        output_dir = os.path.join(UITester._BASE_PATH, folder_name)
+        create_path(output_dir)
+        output_dir = os.path.join(output_dir, self._test_name)
+        create_path(output_dir)
+        output_dir = os.path.join(output_dir, self._get_browser())
+        create_path(output_dir)
+        return output_dir
+
+    def _get_browser(self):
+        return self._driver.capabilities['browserName']
+
+
+class WorksheetTest(UITester):
+    def __init__(self):
+        super().__init__('worksheet')
+
+    def test(self):
+        self.login()
+        self.wait_until_worksheet_loads()
+        self.click_link('Small Worksheet [cl_small_worksheet]')
```

Comment:
Where in the code is the "Small Worksheet" created?

Reply:
It's in `create_sample_worksheet.py`. Because of the changes in https://github.com/codalab/codalab-worksheets/pull/2002, it will already be available in the Travis  environment.